### ISSUE TITLE
BlockVariationPicker visible text has useless ARIA role presentation

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#45753]
### Description

Some blocks use the `BlockVariationPicker` to visually present variations, for example the Columns and Query loop blocks.

The visible text below the variation buttons is placed within a `<span>` element with an ARIA `role=presentation`. This role doesn't do anything, as it's only meant to remove native roles, if any. A span element doesn't have native role.

If the original intent was to hide this text from assistive technologies, an `aria-hidden="true" should be used instead. However, I'd recommend to keep the text available to assistive technologies, as it may add some value. Worst thing can happen is some redundant information.

Introduced in https://github.com/WordPress/gutenberg/pull/19789

From the [ARIA 1.2 spec](https://www.w3.org/TR/wai-aria-1.2/#presentation):

> presentation role
> An element whose implicit native role semantics will not be mapped to the accessibility API. See synonym [none].

> For any element with a role of presentation and which is not focusable, the user agent MUST NOT expose the implicit native semantics of the element (the role and its states and properties) to accessibility APIs. However, the user agent MUST expose content and descendant elements that do not have an explicit or inherited role of presentation. Thus, the presentation role causes a given element to be treated as having no role or to be removed from the accessibility tree, but does not cause the content contained within the element to be removed from the accessibility tree.

> NOTE
> In ARIA 1.1, the working group introduced [none](https://www.w3.org/TR/wai-aria-1.2/#none) as a synonym to the presentation role, due to author confusion surrounding the intended meaning of the word "presentation" or "presentational." Many individuals erroneously consider role="presentation" to be synonymous with aria-hidden="true", and we believe role="none" conveys the actual meaning more unambiguously.

### Step-by-step reproduction instructions

- Add a Columns or Query loop block
- Note: to make the block variations appear in the Query loop block, click 'Start empty'.
- Inspect the source of the visible text below the variation buttons.
- observe the `role="presentation"` attribute is present.
- Optionally test with a screen reader.
- Navigate the variations using the screen reader arrows navigation (ie. with VoiceOver it's Control+Option+Right (or Left) arrow)
- Observe the visible text is read out.

### Screenshots, screen recording, code snippet

Query loop block:

<img width="928" alt="Screenshot 2022-11-14 at 14 43 45" src="https://user-images.githubusercontent.com/1682452/201684156-fb82b97a-3117-4dab-83de-c8544e3f0306.png">

Columns block:

<img width="695" alt="Screenshot 2022-11-14 at 14 55 42" src="https://user-images.githubusercontent.com/1682452/201684195-b9489cc0-fb8a-4020-bbac-cd2fe1afaa41.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by marissa-makes** [comment]
I can reproduce this, and agree that the attribute seems unnecessary. I'll also second the idea of: 

> However, I'd recommend to keep the text available to assistive technologies, as ti may add some value. 

I just opened #45916 for consideration, which simply edits `BlockVariationPicker` markup to remove the extra line, but does not attempt to hide the text from assistive technologies 😄

**Comment by alexstine** [comment]
Thanks @marissa-makes for the PR!

@afercia I am not worried about duplicate text. It is using `label` on the `Button` component which translates to this.

packages/components/src/Button/index.js
`aria-label={ additionalProps[ 'aria-label' ] || label }`

The inner span text will likely never be read. May need a follow-up if you think it could be important. I personally find the default description to work fine but it does fallback to title.

**Comment by alexstine** [comment]
@afercia Sorry, ignore me. That span falls out of the button tag so it will be read if user explores the content. Probably not worth using `aria-describedby` here because that is simply the non-descriptive title. Still, if you think it deserves a follow-up, feel free to open a new issue to restructure it.
- `aria-label` for title.
- `aria-describedby` for description. However, description is not shown visually, I think it is only in a tool tip on hover. This would create a bit of work. After the React upgrade, `aria-description` would become usable.

**Comment by afercia** [comment]
No problem @alexstine 

I wouldn't use aria-describedby for this text until its usage is clarified and improved. For example, in the Query loop this text just repeats the aria-label. Using it for the description would be redundant. In the Columns block instead, it's used as a sort of descriptive text, even though I'm not sure it's that meaningful.
